app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt

```diff
@@ -153,4 +168,31 @@ class SimilarEventsFragment : Fragment() {
         moreLikeThis.isGone = similarEvents.isEmpty()
         similarEventsRecycler.isGone = similarEvents.isEmpty()
     }
+
+    private fun setUpAdapter() {
+        similarEvents.clear()
+        var id: Long
+        var add: Boolean
+        if (similarIdEvents.size != 0 && similarLocationEvents.size != 0) {
+            similarIdEvents.forEach {
+                id = it.id
+                add = true
+                similarLocationEvents.forEach {
+                    if (id == it.id) add = false
```

Comment:
You could add `break` when `id` is matched 
or
 you would like to use kotlin stdlib `find`  for better readability
```
val notfound = list.find{ id == it.id}==null
if(notFound) addtolist 
```

Reply:
resolved